One step further. Now OS X is complaing about `--no-undefined`. I will take a look at this tomorrow.

``` shell
gfortran -fPIC -m64 -shared -Wl,--no-undefined piecewise_subs.o piecewise3.o ../ef_utility/copy_ferret_ef_mem_subsc.o -L/Users/travis/miniconda3/envs/_build_placehold_placehold_placehold_placehold_pla/lib/python2.7/site-packages/pyferret -lpyferret -o piecewise3.so
ld: unknown option: --no-undefined
collect2: error: ld returned 1 exit status
make[4]: *** [piecewise3.so] Error 1
make[3]: *** [build] Error 1
make[2]: *** [optimized] Error 2
make[1]: *** [externals_optimized] Error 2
make: *** [optimized] Error 2
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -7,7 +7,7 @@ os: osx
 
 env:
   matrix:
-    
+
     - CONDA_PY=27
   global:
     # The BINSTAR_TOKEN secure variable. This is defined canonically in conda-forge.yml.
@@ -32,6 +32,8 @@ install:
       conda config --set show_channel_urls true
       conda install --yes --quiet conda-forge-build-setup
       source run_conda_forge_build_setup
+      # Bypass the control-freak run_conda_forge_build_setup and use versions that do work.
+      conda install --yes conda-build=1.21.11
 
 script:
   - conda build ./recipe

--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -32,6 +32,7 @@ if [ $(uname) == Darwin ]; then
     export MACOSX_DEPLOYMENT_TARGET="10.9"
     export CXXFLAGS="-stdlib=libc++ $CXXFLAGS"
     export CXXFLAGS="$CXXFLAGS -stdlib=libc++"
+    export DYLD_FALLBACK_LIBRARY_PATH=$PREFIX/lib
 fi
 
 export LIBZ_DIR=$PREFIX

--- a/recipe/linux-platform_specific.patch
+++ b/recipe/linux-platform_specific.patch
@@ -1,0 +1,12 @@
+--- external_functions/ef_utility/platform_specific.mk.x86_64-linux	2016-08-26 21:50:00.000000000 -0300
++++ external_functions/ef_utility/platform_specific.mk.x86_64-linux	2016-09-07 11:05:27.418727344 -0300
+@@ -23,8 +23,7 @@
+ LD		= gfortran
+ LD_DYN_FLAGS	= -fPIC -m64 -shared -Wl,--no-undefined
+ SYSLIBS		= ../ef_utility/copy_ferret_ef_mem_subsc.o \
+-		  -L$(INSTALL_FER_DIR)/lib64/$(PYTHON_EXE)/site-packages/pyferret \
+-		  -L$(INSTALL_FER_DIR)/lib/$(PYTHON_EXE)/site-packages/pyferret \
++		  -L$(SP_DIR)/pyferret \
+ 		  -lpyferret
+ 
+ # Directly compile the .F source files to the .o object files

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -10,7 +10,9 @@ source:
     url: https://github.com/NOAA-PMEL/PyFerret/archive/{{ commit }}.tar.gz
     sha256: e702f4124cfa3fcfa898f5ec6075c548a5b958c70f99f60faa2119f325fe2127
     patches:
-        - site_specific.patch
+        - site_specific.patch  # [not win]
+        - linux-platform_specific.patch  # [linux]
+        - osx-platform_specific.patch  # [osx]
 
 build:
     number: 4
@@ -39,7 +41,7 @@ test:
         - pyferret
     commands:
         - conda inspect linkages -n _test pyferret  # [not win]
-        - conda inspect objects -n _test nco  # [osx]
+        - conda inspect objects -n _test pyferret  # [osx]
 
 about:
     home: http://ferret.pmel.noaa.gov/Ferret

--- a/recipe/no_no-undefined.patch
+++ b/recipe/no_no-undefined.patch
@@ -1,0 +1,16 @@
+--- external_functions/ef_utility/platform_specific.mk.x86_64-linux	2016-08-26 21:50:00.000000000 -0300
++++ external_functions/ef_utility/platform_specific.mk.x86_64-linux	2016-09-06 15:46:38.895326249 -0300
+@@ -21,11 +21,8 @@
+ FFLAGS_DEBUG	= -O0 -g -fbounds-check 
+ 
+ LD		= gfortran
+-LD_DYN_FLAGS	= -fPIC -m64 -shared -Wl,--no-undefined
+-SYSLIBS		= ../ef_utility/copy_ferret_ef_mem_subsc.o \
+-		  -L$(INSTALL_FER_DIR)/lib64/$(PYTHON_EXE)/site-packages/pyferret \
+-		  -L$(INSTALL_FER_DIR)/lib/$(PYTHON_EXE)/site-packages/pyferret \
+-		  -lpyferret
++LD_DYN_FLAGS	= -fPIC -bundle -undefined dynamic_lookup
++SYSLIBS		= ../ef_utility/copy_ferret_ef_mem_subsc.o
+ 
+ # Directly compile the .F source files to the .o object files
+ # since gfortran can handle the C compiler directives in Fortran code

--- a/recipe/osx-platform_specific.patch
+++ b/recipe/osx-platform_specific.patch
@@ -1,0 +1,16 @@
+--- external_functions/ef_utility/platform_specific.mk.x86_64-linux	2016-08-26 21:50:00.000000000 -0300
++++ external_functions/ef_utility/platform_specific.mk.x86_64-linux	2016-09-07 11:20:01.299770866 -0300
+@@ -21,11 +21,8 @@
+ FFLAGS_DEBUG	= -O0 -g -fbounds-check 
+ 
+ LD		= gfortran
+-LD_DYN_FLAGS	= -fPIC -m64 -shared -Wl,--no-undefined
+-SYSLIBS		= ../ef_utility/copy_ferret_ef_mem_subsc.o \
+-		  -L$(INSTALL_FER_DIR)/lib64/$(PYTHON_EXE)/site-packages/pyferret \
+-		  -L$(INSTALL_FER_DIR)/lib/$(PYTHON_EXE)/site-packages/pyferret \
+-		  -lpyferret
++LD_DYN_FLAGS	= -fPIC -bundle -undefined dynamic_lookup
++SYSLIBS		= ../ef_utility/copy_ferret_ef_mem_subsc.o 
+ 
+ # Directly compile the .F source files to the .o object files
+ # since gfortran can handle the C compiler directives in Fortran code

--- a/recipe/site_specific.patch
+++ b/recipe/site_specific.patch
@@ -1,16 +1,3 @@
-diff -Naur PyFerret-67aed0fa29f2d3f28e624030d746d23d146fc387.orig/external_functions/ef_utility/platform_specific.mk.x86_64-linux PyFerret-67aed0fa29f2d3f28e624030d746d23d146fc387/external_functions/ef_utility/platform_specific.mk.x86_64-linux
---- external_functions/ef_utility/platform_specific.mk.x86_64-linux	2016-06-29 20:09:51.000000000 -0300
-+++ external_functions/ef_utility/platform_specific.mk.x86_64-linux	2016-07-06 14:40:59.260485464 -0300
-@@ -23,8 +23,7 @@
- LD		= gfortran
- LD_DYN_FLAGS	= -fPIC -m64 -shared -Wl,--no-undefined
- SYSLIBS		= ../ef_utility/copy_ferret_ef_mem_subsc.o \
--		  -L$(INSTALL_FER_DIR)/lib64/$(PYTHON_EXE)/site-packages/pyferret \
--		  -L$(INSTALL_FER_DIR)/lib/$(PYTHON_EXE)/site-packages/pyferret \
-+		  -L$(SP_DIR)/pyferret \
- 		  -lpyferret
- 
- # Directly compile the .F source files to the .o object files
 diff -Naur PyFerret-67aed0fa29f2d3f28e624030d746d23d146fc387.orig/external_functions/ef_utility/site_specific.mk PyFerret-67aed0fa29f2d3f28e624030d746d23d146fc387/external_functions/ef_utility/site_specific.mk
 --- external_functions/ef_utility/site_specific.mk	2016-06-29 20:09:51.000000000 -0300
 +++ external_functions/ef_utility/site_specific.mk	2016-07-06 14:34:56.858490288 -0300

